packages/core/saved-objects/core-saved-objects-migration-server-internal/src/zdt/state/create_initial_state.ts

```diff
@@ -13,11 +13,14 @@ import type { MigratorContext } from '../context';
  * Create the initial state to be used for the ZDT migrator.
  */
 export const createInitialState = (context: MigratorContext): State => {
+  const runDocumentMigration =
+    context.nodeRoles.migrator || context.migrationConfig.zdt.runOnNonMigratorNodes;
```

Comment:
As explained on the PR's description: this `zdt.runOnNonMigratorNodes` was added to force non-migrator nodes to run the document migration. The only usage for now is integration tests, but we may want to use this in single-node dev / test environments.